File: content/problems/0357-count-numbers-with-unique-digits/solutions/combinatorial-counting/Solution.java
class Solution {
    public int countNumbersWithUniqueDigits(int n){
        if(n==0)return 1;
        int total=10,unique=9,available=9;
        for(int digits=2;digits<=n;digits++){
            unique*=available--;
            total+=unique;
        }
        return total;
    }
}

File: content/problems/0357-count-numbers-with-unique-digits/solutions/combinatorial-counting/Solution.kt
class Solution {
    fun countNumbersWithUniqueDigits(n:Int):Int{
        if(n==0)return 1;
        var total=10;
        var unique=9;
        var available=9;
        for(digits in 2..n){
            unique*=available--;
            total+=unique
        };
        return total
    }
}
